MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java

```diff
@@ -1563,15 +1565,24 @@ public static Expression in(@NonNull Number needle, @NonNull Expression haystack
   /**
    * Retrieves whether an item exists in an array or a substring exists in a string.
    *
-   * @param needle     the item expression
-   * @param haystack   the array or string expression
+   * @param needle   the item expression
+   * @param haystack the array or string expression
    * @return true if exists.
    * @see <a href="https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-in">Style specification</a>
    */
   public static Expression in(@NonNull String needle, @NonNull Expression haystack) {
     return new Expression("in", literal(needle), haystack);
   }
 
+  public static Expression within(@NonNull Polygon polygon) {
```

Comment:
pretty cool that you got it to work with a geojson type. This is a very handy utility function. To be feature complete we will need to expose `within(@NonNull Expression expression)` so developers can use other expressions in conjunction with this one.

Reply:
update: now with looking more into upstream PR, we do need to expose only the Polygon variation. Expressions will now support inline geometry!